engine/eConstr.ml

```diff
@@ -1020,7 +1020,9 @@ let noccurn sigma n term =
     | _ -> Expand.iter_with_binders sigma succ occur_rec n h knd
   in
   let h, term = Expand.make term in
-  try occur_rec n h term; true with LocalOccur -> false
+  try occur_rec n h term; true with
+  | LocalOccur -> false
+  | _ when !Flags.in_debugger -> false
```

Comment:
This looks like the wrong place to fix. We're failing with an anomaly for good reasons, it should be the job of the printer to be defensive in debugging mode.